email_account_validity/_utils.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import random
+import re
+import string
+
+
+UNAUTHENTICATED_TOKEN_REGEX = re.compile('^[a-zA-Z]{32}$')
+
+# We use SystemRandom to make sure we get cryptographically-secure randoms.
+rand = random.SystemRandom()
+
+
+def random_digit_string(length):
+    return "".join(rand.choice(string.digits) for _ in range(length))
```

Comment:
You can use [`secrets.choice`](https://docs.python.org/3/library/secrets.html#secrets.choice) to always get a CSPRNG

Functionally identical to `random.SystemRandom().choice`, but more obvious that it's secure.